### PR TITLE
Fix navigation of table of content links

### PR DESCRIPTION
**Pull request by callingmedic911**
Closes PRO-254.

This solution isn't as clean as I was hoping. With Next.js 14, we don't have router.events in app router components and we can not directly attach a listener to `hashchange` because the Next router uses `pushState` that doesn't fire this event. I ended up detecting if there is a hash change with `useParams` hooks and then calculating the hash from directly window object.

### CODE REVIEW

File: src/components/TableOfContents.tsx [hunk 1]
```diff
@@ -65,7 +74,7 @@ export function TableOfContents({
   return (
     <div
       data-testid="table-of-contents"
-      className="hidden py-8 xl:sticky xl:-mr-6 xl:block xl:h-[calc(100vh-4.75rem)] xl:flex-none xl:overflow-y-auto xl:pr-6 xl:top-[4.75rem] xl:py-16"
+      className="hidden py-8 xl:sticky xl:top-[4.75rem] xl:-mr-6 xl:block xl:h-[calc(100vh-4.75rem)] xl:flex-none xl:overflow-y-auto xl:py-16 xl:pr-6"
```
Comment (callingmedic911) [inline src/components/TableOfContents.tsx:77]: Unrelated lint change on autosave.

**Review by jonbell-lot23 — APPROVED** [review]
